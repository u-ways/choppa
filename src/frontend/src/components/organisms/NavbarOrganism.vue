<template>
  <nav class="container-fluid navbar">
    <div class="navbar__element ml-1 ml-sm-3 navbar__link">
      <router-link to="/">
        <div class="justify-content-start">
          <div class="d-inline-block pr-1">
            <ChoppaLogoAtom css="navbar__logo"/>
          </div>
          <div class="navbar__name d-inline-block">
            Choppa<span class="navbar__name--light">.app</span>
          </div>
        </div>
      </router-link>
    </div>
  </nav>
</template>

<script>
import ChoppaLogoAtom from "@/components/atoms/ChoppaLogoAtom";

export default {
  components: { ChoppaLogoAtom },
};
</script>

<style lang="scss">
@import "src/assets/scss/colours";
@import "src/assets/scss/typography";

.navbar {
  margin: 0;
  padding: 0;
  background: $white;
  box-shadow: 0 2px 6px 0 hsla(0, 0%, 0%, 0.2);

  &__element {
    padding: 0.8rem 0.5rem 0.2rem;
    border-bottom: 3px solid hsla(0, 0, 0, 0);
  }

  &__link {
    &:hover,
    &:focus {
      border-bottom-color: $accent;
      transition: 0.4s border-bottom-color;
    }
  }

  &__logo {
    font-size: 2rem;
    color: $text-dark;
  }

  &__name {
    font-size: 1.45rem;
    font-weight: $font-weight-medium;
    color: $dark-grey;

    &--light {
      font-weight: $font-weight-light;
    }
  }
}
</style>
