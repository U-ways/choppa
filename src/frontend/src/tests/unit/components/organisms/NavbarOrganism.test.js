import { createLocalVue, describe, expect, it, shallowMount } from "@/tests/unit/test-imports";
import VueRouter from "vue-router";
import NavbarOrganism from "@/_old/components/organisms/NavbarOrganism";

describe("Navbar Organism test", () => {
  it("Renders correctly", async () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();

    const wrapper = shallowMount(NavbarOrganism, {
      localVue,
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
