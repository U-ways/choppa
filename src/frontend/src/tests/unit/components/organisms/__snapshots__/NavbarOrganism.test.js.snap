// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Organism test Renders correctly 1`] = `
"<nav class=\\"container-fluid navbar\\">
  <router-link-stub to=\\"/\\" tag=\\"a\\" activeclass=\\"navbar__element ml-1 ml-sm-3 navbar__link\\" ariacurrentvalue=\\"page\\" event=\\"click\\">
    <div class=\\"justify-content-start\\">
      <div class=\\"d-inline-block pr-1\\">
        <choppalogoatom-stub css=\\"navbar__logo\\"></choppalogoatom-stub>
      </div>
      <div class=\\"navbar__name d-inline-block\\">
        Choppa<span class=\\"navbar__name--light\\">.app</span></div>
    </div>
  </router-link-stub>
</nav>"
`;
